feat: Add AKS First platform guide to docs (#1063)
meshcloud/meshcloud-docs: docs/meshstack.aks-platform-quickstart-guide.md

---
id: meshstack.aks-platform-quickstart-guide
title: AKS Platform Quickstart Guide
---

# Introduction

Welcome to the **First Platform: AKS guide**! This guide provides step-by-step instructions to help you quickly set up
your first meshStack platform integration with Azure Kubernetes Service (AKS).

# Goals

By the end of this guide, you will have achieved:

1. Creating a platform integration with AKS. This will allow:
   - provisioning new namespaces
   - assign users
   - assign tags (optional)
2. Successfully testing the AKS integration

> This allows your internal users to open up meshStack and provision themselves namespaces, including identity & access management.

Reviewer: ```suggestion
This allows your users to open up meshStack and provision themselves namespaces and authorization on AKS.
```
You could reference application teams to make it clearer. The internal user should be clear from context of the docs

## Prerequisites

Before starting, ensure you have the following:

- An AKS Cluster that is publicly accessible via the internet
- A place to run Terraform. For example, your local machine, or the Azure cloud shell.

## Preparing Infrastructure in Azure

To integrate your AKS cluster into meshStack properly, we require a few things to be in place in AKS & Azure for a successful integration.

In this guide, we will use our Terraform module to set these things up. You can also do this manually, but we highly recommend doing this with Terraform, and for this guide we assume you did.

Amongst other things, the Terraform module will create a service user with the right permissions in the Kubernetes cluster and create a service principal in Azure to interact with your Entra tenant.

Click [here](https://github.com/meshcloud/terraform-aks-meshplatform) and follow along the guide in the README to use the Terraform module.

Make sure you execute the last step of exporting the outputs and save these somewhere. We need them later in this guide.

## Creating the Platform

> 1. **Open the platform list in the Platform Builder**

Reviewer: ```suggestion
1. **Open the platform list in the platform builder**
```

> 1. **Open the platform list in the service management area**

Reviewer: ```suggestion
1. **Open the platform list in the service management area**
```
Starting with the release today this will be called platform builder

>    When entering your workspace, click the blue “Go to Services Management” button at the top-right if you haven’t already. Then open the menu on the left-hand side and click on “Platform → Platforms”.

Reviewer: ```suggestion
   When entering your workspace, click the blue “Platform Builder” button at the top-right if you haven’t already. Then open the menu on the left-hand side and click on “Platform → Platforms”.
```

2. **Create a new platform**

   In the list, click on the button “+ Create new platform”. A wizard will open where we will have to enter some information.

3. **Select platform type**

   Under “Platform Type” pick “Azure Kubernetes Service”.

4. **Enter name, identifier & description**

   Enter a fitting name, for example “Azure Kubernetes Service”. You can pick a more specific name if you have multiple clusters for example.

   For the identifier pick something similar to the name, for example `aks`.

   For the description you can write a little bit of additional information about your cluster that people can see in the marketplace. You can also leave this empty for now.

> 5. **Enter Cloud Platform Web Console URL**

Reviewer: ```suggestion
5. **Enter cloud platform web console URL**
```

   Enter a URL here that users can use to open the web console of your AKS cluster. You can enter the API Server Address that can be found in Azure Portal when opening up your AKS cluster.

   The support & documentation fields below can be left empty.

6. **Create a new location**

   Locations are a grouping mechanism for multiple platforms in meshStack. For now you can create a new location and give it the name and identifier “AKS”. After that, click “Next”.

7. **Pricing**

   You will be prompted to enter a price for your new platform. You can skip this for now.

🎉 The platform is created in meshStack! But it won’t be usable yet. We will have to properly integrate it now

# Configuring Integration

> 1. **Open the Configuration**

Reviewer: Please fix capitalization. The list above is all small besides the first word the list below uses capitalization used in headings

   In the last step you should have landed on the configuration page. If not, or if you navigated away again in the meantime, open up the platform and go to “Settings → Configuration”.

2. **Enter base URL**

   Enter the base URL for calling the APIs of your cluster. An example URL is `https://aks-meshcloud-dev-dns-95df19f5.hcp.westeurope.azmk8s.io`.

3. **Add replication config**

   Click the green “+ Add config” button under “Replication Configuration”. This configuration is responsible for managing namespaces & IAM.

4. **Enter access token**

   From your Terraform outputs from before (Preparing Infrastructure step) grab the `replicator_expose_token` value and paste it into the “Access Token” field.

   The next field “Namespace Name Pattern” you can leave as-is for now.

5. **Configure Entra Tenant**

>    In the section below under “Additional Access Configuration” go AKS-specific configuration so that meshStack can manage user & groups in your Entra tenant.

Reviewer: Maybe it improves readability when we provide bullet points when there are more actions required to finish a step

   After that is done, you can look up the `replicator_service_principal` output of the Terraform module and copy the value of the property `Application_Client_ID` into the field in meshStack called “Application (client) ID”.

   In the same object, you should also find the property `Enterprise_Application_Object_ID`. The value of this should go into the field “Enterprise Application Object ID”.

   **Credentials**

   Select “Credentials” for Authentication Type. Make sure you also set `create_password = true` in the meshPlatform AKS Terraform module as a new password is needed for the Service Principal.

   After that is done, you can look up the `replicator_service_principal` output of the Terraform module and copy the value of the property `Application_Client_ID` into the field in meshStack called “Application (client) ID”.

   In the same object you should also find the property `Enterprise_Application_Object_ID`. The value of this should go into the field “Enterprise Application Object ID”.

   Lastly, enter the value of `Client_Secret` into the field “Client Secret”.

> 6. **Cluster Configuration**

Reviewer: ```suggestion
6. **Cluster configuration**
```

   You can leave “User Group Name Pattern” as is. If you configured an Administrative Unit for the service principal then enter its ID in the field “Administrative Unit ID”.

   Enter the subscription ID where the AKS cluster is hosted in “Subscription ID”.

   Enter the name of the AKS cluster in “Cluster Name”.

   Enter the Resource Group name where the AKS cluster is hosted in “Resource Group”.

   You can leave User Lookup Strategy as is and you can leave the B2B User Federation section untouched.

   We will also skip the “Metering Configuration” part of this guide as it is related to monitoring costs and setting prices of your cluster.

🎉 All the necessary configuration has been filled in! Let’s create a landing zone now and then test the new platform.

## Creating a Landing Zone

The last step before testing the platform is creating a so-called landing zone. The landing zone contains configuration for auto-deletion settings, role mappings & more. Landing zones can be used to split up different behavior depending on environments e.g. dev & prod. It is mandatory for a platform to have a landing zone.

> 1. **Create new Landing Zone**

Reviewer: ```suggestion
1. **Create new landing zone**
```

   Open up the “Landing Zones” tab in your platform and click “+ Create new Landing Zone”.

2. **Fill in general information**

   Enter a fitting name, identifier and description for the landing zone. You can simply enter “AKS” for name and identifier and enter a simple description such as “Default setup for a new AKS namespace.

3. **Enable deletion**

   If wanted, you can enable automated tenant deletion by checking both boxes under “Tenant Deletion”. This will allow your application teams to delete their namespaces in self-service using meshStack.

4. **Enter role mappings**

   On the right-hand side create a total of three role mappings by clicking “+ Add Role Mapping” three times.

   In here we can select which role in meshStack relates to which role in your kubernetes cluster. As a start, we recommend the following mapping:

   - Project Admin → `admin`
   - Project User → `edit`
   - Project Reader → `view`

> 5. **Create the Landing Zone**

Reviewer: ```suggestion

5. **Create the Landing Zone**
```

   Below the role mappings is a section called Quotas. We will skip this for now, but it allows you to manage default quotas for newly created namespaces.

   Finish the process by clicking “Save” at the bottom of the page.

🎉 The landing zone has been created! Let’s get to testing

# Testing Tenant Creation

> We will now use your same workspace to test the flow of creating a new namespace (also known as a meshStack **tenant**) from the completely same perspective as your application teams.

Reviewer: ```suggestion
We will now use the same workspace to test the flow of creating a new namespace (also known as a meshStack **tenant**) from the completely same perspective as your application teams.
```
